feat(csi-node): parse and use keep-alive-tmo when connecting to a target

tiagolobocastro opened the pull request:
Adds nvme-keep-alive-tmo via the cmdline which is used as a default for all targets. Alternatively, it can also be overridden via the storage class with nvmeKeepAliveTmo.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
feat(csi-node): parse and use keep-alive-tmo when connecting to a target

Adds nvme-keep-alive-tmo via the cmdline which is used as a default for all targets.
Alternatively, it can also be overridden via the storage class with nvmeKeepAliveTmo.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

Abhinandan-Purkait commented:
Since this can be set via the storage class, should we add this as a part of the publish context as well so that we can use it in HA?

avishnu commented:
nitpick: Should the string have Timeout instead of abbr Tmo?

tiagolobocastro commented:
> Since this can be set via the storage class, should we add this as a part of the publish context as well so that we can use it in HA?

Ah good point, done!

tiagolobocastro committed:
refactor(csi-controller): reduce code duplication for params

Reuse publish params in volume creation param validation.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
test: fix failing test due to race

Also removes missing debug log message..

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: don't use path env during build

Avoids a bug where rustup adds cargo bin twice:
https://github.com/rust-lang/rustup/pull/2978

For all intents and purposes we should be fine to use the current path from runtime.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge
